data-structures-csharp/data-structures-csharp/BinarySearchTree/BinarySearchTree.cs

```diff
@@ -415,6 +415,37 @@ private bool IsInRange(TKey start, TKey end, Node<TKey, TValue> node)
         }
 
         /// <summary>
+        /// Traverse through all the nodes from root node and write the data in Pre-Order Traversal. This returns the result in ascending order
+        /// </summary>
+        /// <param name="root"></param>
+        /// <returns></returns>
+        private List<KeyValuePair<TKey, TValue>> GetAllNodesInPreOrder(Node<TKey, TValue> root)
+        {
+            Contract.Requires<ArgumentNullException>(root != null);
+            Contract.Requires<ArgumentNullException>(this.root != null);
+            Contract.Ensures(Contract.Result<List<KeyValuePair<TKey, TValue>>>() != null);
+
+            var list = new List<KeyValuePair<TKey, TValue>>();
+
+            if (root.Equals(new NullNode<TKey, TValue>()))              //If root node is a NullNode
```

Comment:
It creates a `NullNode` object every time, better create a static `NullNode` object member for the class, that way you don't pay the penalty of creating new object everytime.